Updates for WP Privacy integration

garretthyder opened the pull request:
## Description
Adds support for group_description for privacy exporters which was added in WP5.3 through [WPCoreChangeset#45825](https://core.trac.wordpress.org/changeset/45825) and [WPCoreTracTicket#45491](https://core.trac.wordpress.org/ticket/45491)

Update Suggested Privacy Policy text to utilize privacy-policy-tutorial css class instead of wp-policy-help as it was deprecated. Also wrap the contents in the wp-suggested-text div to style the section to match WordPress. This follows from [WPCoreTrac#49282](https://core.trac.wordpress.org/ticket/49282) and although back-compat is being introduced in WP5.4 as of [WPChangeset#47112](https://core.trac.wordpress.org/changeset/47112) this change will better support users of WP5.1-5.4

This also removes the wrapping <div contenteditable="false"> as it's now unnecessary. As long as the content uses the proper privacy-policy-tutorial class then the WP Privacy Policy Guide section copy action will avoid copying that content. The contenteditable divs were there originally when the contents was fully copied due to not using the appropriate class, and when originally the entire guide would be dumped into the WYSIWYG editor with wp-policy-help sections being highlighted.

## How has this been tested?
I tested this by pulling up the privacy policy guide to confirm the styling was fixed and the contents of the clipboard when copied doesn't contain the tutorial test.
For the exporter I setup a customer with some basic data and ran the exporter to ensure the group descriptions were added properly.

## Types of changes
Bug Fix - The privacy policy guide text is now styled appropriately and the tutorial text is no longer present in the copied contents.
Enhancements - The exporters were updated to include a group description.

## Checklist:
- [x] My code has been tested.
- [x] My code passes all existing automated tests. <!-- Check code: `composer run-script tests-run`, Guidelines: https://github.com/gocodebox/lifterlms/blob/master/tests/README.md -->
- [x] My code follows the LifterLMS Coding & Documentation Standards. <!-- Check code: `composer run-script phpcs`, Guidelines: https://github.com/gocodebox/lifterlms/blob/master/docs/coding-standards.md and https://github.com/gocodebox/lifterlms/blob/master/docs/documentation-standards.md -->

thomasplevy commented on includes/privacy/class-llms-privacy-exporters.php:
```diff
@@ -566,14 +575,16 @@ public static function quiz_data( $email_address, $page ) {
 
-			$group_label = __( 'Quiz Attempts', 'lifterlms' );
+			$group_label        = __( 'Quiz Attempts', 'lifterlms' );
+			$group_descriptions = __( 'Student quiz attempts data', 'lifterlms' );
```
Typo: shouldn't be plural

thomasplevy commented on includes/privacy/class-llms-privacy.php:
```diff
@@ -130,46 +130,44 @@ protected static function get_order_data_props( $type ) {
-				'<p class="wp-policy-help">' . __( 'In this section you should list who you’re sharing data with, and for what purpose. This could include, but may not be limited to, analytics, marketing, payment gateways, and third party embeds.', 'lifterlms' ) . '</p>' .
-			'</div>' .
-			'<p>' . __( 'We share information with third parties who help us provide our orders and store services to you; for example --', 'lifterlms' ) . '</p>';
+   			    '<p>' . __( 'We collect information about you during the registration, enrollment, and checkout processes on our site.', 'lifterlms' ) . '</p>' .
```
Whitespace error: run `composer run-script fix-cs` or manually replace spaces with tabs.

garretthyder commented:
No worries @thomasplevy I've fixed the two minor issues you flagged and added `@since 3.37.8` for the changelog. Is that the correct versioning to use? Also should I update the @version or is that the original version that function was added in?

thomasplevy commented:
@garretthyder thanks!

We actually use use "[version]" as a placeholder when updating files and replace the placeholder with the actual version number during a build step when packaging releases. Docs: https://github.com/gocodebox/lifterlms/blob/master/docs/documentation-standards.md#using-placeholders

And we've been phasing out the `@version` placeholders over time as they're redundant and unnecessary but removing every instance from the codebase all at once is tiresome and a waste of our time. We remove them as we update files. But don't worry about that, I'll get those removed when I merge it in (and i'll swap the 3.37.8 for the version since this will be included in 3.37.9 (or maybe 3.38).

Thanks again!

thomasplevy commented:
I'm not sure why so many of the tests are failing now... I'll have to pull this down to test locally but that's not going to happen for a few days. We're getting ready to ship a new plugin early next week and non-emergency updates to the core are being held until after that release.

garretthyder commented:
Thanks @thomasplevy ping me if you need me to make a refresh in anyway. Looking at the failing tests it's a timezone issue or appears to be which is unrelated to my changes.

Error;
```
1) LLMS_Test_Payment_Gateway_Integrations::test_recurring_lifecycle_for_month_plan_with_chaos
Month Payment #21: Got 2021-10-30 20:31:51 and expected 2021-11-01 04:31:47
Failed asserting that 1635625911 matches expected 1635741107.
```

And no rush from me, all the best

thomasplevy commented:
Yea... I'm just not sure why that would suddenly start failing on this PR but isn't on master or my local... I'll look into it

thomasplevy commented:
Closing in favor of #1021
